renku/service/controllers/cache_files_upload.py

```diff
@@ -0,0 +1,116 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Renku service cache upload files controller."""
+import os
+import shutil
+from pathlib import Path
+
+import patoolib
+from patoolib.util import PatoolError
+
+from renku.core.errors import RenkuException
+from renku.service.config import CACHE_UPLOADS_PATH, SUPPORTED_ARCHIVES
+from renku.service.controllers.api.abstract import ServiceCtrl
+from renku.service.controllers.api.mixins import ReadOperationMixin
+from renku.service.serializers.cache import FileUploadRequest, FileUploadResponseRPC, extract_file
+from renku.service.views import result_response
+
+
+class UploadFilesCtrl(ServiceCtrl, ReadOperationMixin):
+    """Controller for migrations check endpoint."""
+
+    REQUEST_SERIALIZER = FileUploadRequest()
+    RESPONSE_SERIALIZER = FileUploadResponseRPC()
+
+    def __init__(self, cache, user_data, flask_request):
+        """Construct controller."""
+        self.file = extract_file(flask_request)
+
+        self.response_builder = {
+            "file_name": self.file.filename,
+            "content_type": self.file.content_type,
+            "is_archive": self.file.content_type in SUPPORTED_ARCHIVES,
+        }
+        self.response_builder.update(UploadFilesCtrl.REQUEST_SERIALIZER.load(flask_request.args))
+
+        super(UploadFilesCtrl, self).__init__(cache, user_data, {})
+
+    @property
+    def context(self):
+        """Controller operation context."""
+        return self.response_builder
+
+    def process_file(self):
+        """Process uploaded file."""
+        user_cache_dir = CACHE_UPLOADS_PATH / self.user.user_id
+        user_cache_dir.mkdir(exist_ok=True)
+
+        file_path = user_cache_dir / self.file.filename
+        if file_path.exists():
+            if self.response_builder.get("override_existing", False):
+                file_path.unlink()
+            else:
+                raise RenkuException("file exists")
+
+        self.file.save(str(file_path))
+
+        files = []
+        if self.response_builder["unpack_archive"] and self.response_builder["is_archive"]:
+            unpack_dir = "{0}.unpacked".format(file_path.name)
+            temp_dir = file_path.parent / Path(unpack_dir)
+            if temp_dir.exists():
+                shutil.rmtree(str(temp_dir))
+            temp_dir.mkdir(exist_ok=True)
+
+            try:
+                patoolib.extract_archive(str(file_path), outdir=str(temp_dir))
+            except PatoolError:
+                return RenkuException("unable to unpack archive")
+
+            for file_ in temp_dir.glob("**/*"):
+                relative_path = file_.relative_to(CACHE_UPLOADS_PATH / self.user.user_id)
+
+                file_obj = {
+                    "file_name": file_.name,
+                    "file_size": os.stat(str(file_path)).st_size,
```

Comment:
Yes. It's not always an archive. This field size indicates the uploaded data size and its behavior does not depend on the provided `unpack_archive` attribute in the request. Behaviour is consistent for both cases. I agree though it might be useful to add `unpacked_size` as well in this case though. But this might be slightly out of the scope here, so until we find a good use case for it I would wait with this modification (also, we would need to sync with the UI for it).

Reply:
I think the initial design was that field contains the size of the uploaded file and I'm not sure how this evolved to be like this. This is definitely looking off. Fixing it.